review_practice/react-app/src/components/PersonnelController/index.js

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect } from 'react';
+
+import * as Style from './style';
+
+/* 
+personnelItems: [state, setState, rendering되는 이름, payload에 담길 이름]
+personnelControlled: 버튼클릭 시 호출되는 함수
+*/
+export default ({ personnelItems, requestToServer, personnelControlled, deleteButtonHandle }) => {
+    const saveButtonHandler = () => () => {
+        const personnel = personnelItems.reduce((payload, item) => {
+            const [stateValue, stateName] = [item[0], item[3]];
+            payload[stateName] = stateValue;
+            return payload;
+        }, {});
+        requestToServer(personnel);
+    };
+
+    const decreaseButtonHandler = (state, setState) => () => {
+        if (state === 0) return;
+        setState(state - 1);
+    };
+
+    const increaseButtonHandler = (state, setState) => () => {
+        setState(state + 1);
+    };
+
+    useEffect(() => {
+        const states = personnelItems.map(item => item[0]);
+        personnelControlled(states);
+    });
+
+    return (
+        <Style.PersonnelController>
+            {personnelItems.map((state, index) => (
```

Comment:
이걸 state라고 퉁쳐서 받기에는 조금 어색한 것 같습니다.
```suggestion
            {personnelItems.map(([state, setState], index) => (
```
이렇게 해체하여 받거나 혹은 더 명시적인 이름으로 선언해주면 어떨가요?